lib/Conversion/TTIRToTTMetal/TTIRToTTMetal.cpp

```diff
@@ -741,36 +750,137 @@ class TTIRToTTMetalDispatchRewriter : public OpRewritePattern<ttir::GenericOp> {
   }
 
   void convertComputeBinaryOp(Operation &arithOrMathOp,
                               ArrayRef<BlockArgument> cbOperands,
                               ArrayRef<BlockArgument> iterators,
                               SmallVector<unsigned> blockArgIteratorMapping,
-                              Value dstTileIndex, OpBuilder &builder) const {
+                              OpBuilder &builder) const {
     assert(cbOperands.size() == 3 &&
            "Expected two input and one output CB for binary op.");
 
     auto inCB0TileIndex = iterators[blockArgIteratorMapping[0]];
     auto inCB0 = cbOperands[0];
     auto inCB1TileIndex = iterators[blockArgIteratorMapping[1]];
     auto inCB1 = cbOperands[1];
+    auto outCB = cbOperands[2];
 
     // Perform computation C = A (*) B on tile A from inCB0 and tile B from
     // inCB1 and store the result C in DST register on dstTileIndex.
+
     if (mlir::isa<arith::AddFOp>(arithOrMathOp)) {
+      Value dstIndex = i32(0, builder);
+      builder.create<ttkernel::TileRegsAcquireOp>(arithOrMathOp.getLoc());
       builder.create<ttkernel::AddTilesOp>(arithOrMathOp.getLoc(), inCB0, inCB1,
                                            inCB0TileIndex, inCB1TileIndex,
-                                           dstTileIndex);
+                                           dstIndex);
+      builder.create<ttkernel::TileRegsCommitOp>(arithOrMathOp.getLoc());
+      builder.create<ttkernel::TileRegsWaitOp>(arithOrMathOp.getLoc());
+      builder.create<ttkernel::PackTileOp>(
+          arithOrMathOp.getLoc(), dstIndex, outCB,
+          iterators[blockArgIteratorMapping[2]]);
+      builder.create<ttkernel::TileRegsReleaseOp>(arithOrMathOp.getLoc());
     } else if (mlir::isa<arith::MulFOp>(arithOrMathOp)) {
-      builder.create<ttkernel::MulTilesOp>(arithOrMathOp.getLoc(), inCB0, inCB1,
-                                           inCB0TileIndex, inCB1TileIndex,
-                                           dstTileIndex);
+      commonComputeMulOp(arithOrMathOp, cbOperands, iterators,
+                         blockArgIteratorMapping, builder);
+    } else if (mlir::isa<arith::DivFOp>(arithOrMathOp)) {
+
+      SmallVector<std::int64_t> operandIndicesRecip;
+      // For DIV, input 1 is going through reciprocal.
+      operandIndicesRecip.push_back(1);
+      commonComputeRecipOp(arithOrMathOp, cbOperands, iterators,
+                           blockArgIteratorMapping, builder,
+                           operandIndicesRecip);
+
+      builder.create<ttkernel::MulTilesInitOp>(arithOrMathOp.getLoc(), inCB0,
+                                               inCB1);
+
+      commonComputeMulOp(arithOrMathOp, cbOperands, iterators,
+                         blockArgIteratorMapping, builder);
+
+      Value one = i32(1, builder);
+      builder.create<ttkernel::CBPopFrontOp>(arithOrMathOp.getLoc(), inCB1,
+                                             one);
```

Comment:
This does not look nice anymore. We should prioritize https://github.com/tenstorrent/tt-mlir/issues/807.

Reply:
Agree, will prioritize